src/app.rs

```diff
@@ -15,17 +15,23 @@
 use std::net::SocketAddr;
 use std::time::Duration;
 
+use prometheus_client::registry::Registry;
 use tokio::task::JoinHandle;
 use tokio::time;
 use tracing::{error, info, warn};
 
 use crate::identity::CertificateProvider;
+use crate::monitoring::BuildMetrics;
 use crate::{admin, config, identity, proxy, signal, workload};
 
 pub async fn build_with_cert(
     config: config::Config,
     cert_manager: impl CertificateProvider,
 ) -> anyhow::Result<Bound> {
+    let mut registry = Registry::default();
+    let ztunnel_registry = registry.sub_registry_with_prefix("istio");
+    BuildMetrics::register(ztunnel_registry);
```

Comment:
This may work right now but I don't think this works long term; we can't pass `ztunnel_registry` to everyone since it needs to be `mut`. I am looking into some alternatives but probably ok to merge this as-is for now and iterate